staging/src/k8s.io/apiserver/pkg/apis/config/v1/defaults_test.go

```diff
@@ -67,17 +77,88 @@ func TestKMSProviderCacheDefaults(t *testing.T) {
 		{
 			desc: "cache size not supplied",
 			in:   &KMSConfiguration{},
-			want: &KMSConfiguration{Timeout: defaultTimeout, CacheSize: &defaultCacheSize},
+			want: &KMSConfiguration{
+				Timeout:           defaultTimeout,
+				CacheSize:         &defaultCacheSize,
+				CacheHealthyTTL:   defaultCacheHealthyTTL,
+				CacheUnHealthyTTL: defaultCacheUnHealthyTTL,
+			},
 		},
 		{
 			desc: "cache of zero size supplied",
 			in:   &KMSConfiguration{CacheSize: &zero},
-			want: &KMSConfiguration{Timeout: defaultTimeout, CacheSize: &zero},
+			want: &KMSConfiguration{
+				Timeout:           defaultTimeout,
+				CacheSize:         &zero,
+				CacheHealthyTTL:   defaultCacheHealthyTTL,
+				CacheUnHealthyTTL: defaultCacheUnHealthyTTL,
+			},
 		},
 		{
 			desc: "positive cache size supplied",
 			in:   &KMSConfiguration{CacheSize: &ten},
-			want: &KMSConfiguration{Timeout: defaultTimeout, CacheSize: &ten},
+			want: &KMSConfiguration{
+				Timeout:           defaultTimeout,
+				CacheSize:         &ten,
+				CacheHealthyTTL:   defaultCacheHealthyTTL,
+				CacheUnHealthyTTL: defaultCacheUnHealthyTTL,
+			},
+		},
+	}
+
+	for _, tt := range testCases {
+		t.Run(tt.desc, func(t *testing.T) {
+			SetDefaults_KMSConfiguration(tt.in)
+			if d := cmp.Diff(tt.want, tt.in); d != "" {
+				t.Fatalf("KMS Provider mismatch (-want +got):\n%s", d)
+			}
+		})
+	}
+}
+
+func TestKMSProviderHealthCacheTTLDefaults(t *testing.T) {
```

Comment:
How about adding new test cases to TestKMSProviderCacheDefaults instead?

Reply:
TestKMSProviderCacheDefaults handles the cache for DEKs.
I agree that many of these tests could be combined.